apps/meteor/app/lib/server/methods/sendMessage.js

```diff
@@ -107,8 +108,9 @@ Meteor.methods({
 		}
 
 		try {
-			if (Promise.await(federationRoomServiceSender.isAFederatedRoom(message.rid))) {
-				return federationRoomServiceSender.sendMessageFromRocketChat(
+			const room = Rooms.findOneById(message.rid);
```

Comment:
while suggestion to add a projection to return only the field `federated`, I realised a best place for this check is actually after like 78, where `room` is already defined.

Reply:
Makes sense.